packages/console/src/assets/docs/guides/m2m-general/README.mdx

```diff
@@ -22,15 +25,43 @@ Get your app ID and app secret.
 <ApplicationCredentials />
 
 </Step>
-<Step title="Enable admin access" subtitle="(optional)">
+{ isDevFeaturesEnabled ? (
+  <Step title="Enable admin access" subtitle="(optional)">
 
-### Accessing Logto Management API
+  ### Accessing Logto Management API
 
-If you want to use this m2m app to access Logto [Management API](https://docs.logto.io/docs/references/core/#management-api), you will also need to enable "admin access" for you application.
+  If you want to use this m2m app to access Logto [Management API](https://docs.logto.io/docs/references/core/#management-api).
 
-<EnableAdminAccess />
+  You should go to "Roles" tab and create a new machine-to-machine role with the permission of management API.
 
-</Step>
+  <img
+    alt="Create m2m management API role"
+    src={CreateM2mManagementApiRole}
+    width="600px"
+    style={{ borderRadius: '6px' }}
+  />
+
+  And then go to the "Roles" tab under machine to machine app detail page, and assign the role you just created to the app.
+
+  <img
+    alt="Assign m2m management API role to app"
+    src={AssignManagementApiRole}
+    width="600px"
+    style={{ borderRadius: '6px' }}
+  />
+
+  </Step>
```

Comment:
Can we implement a component here and allow the user to create the machine-to-machine role in guide? Otherwise, the user might not be able to follow the rest of the steps in this guide.

Reply:
Since this is a temporary solution and will no longer have this logic after m2m RBAC goes live, I would rather not spend a lot of time on it.

Reply:
Per our offline discussion, reorged the structure of the m2m creation guide, put the "optional" admin access part to the end.